### ISSUE TITLE
WPy64-3940_VSCode not portable

### ISSUE THREAD

**Issue by Axle-Ozz-i-sofT** [winpython/winpython#1002]
I just tested the VSCode and is not portable, Writes files to \User\Name\AppData\Roaming\Python39\*
Attempted both:
\WPy64-3940\VS Code.exe
\WPy64-3940\t\VSCode\Code.exe (which includes the data directory to make VSCode portable)
I have several dev environments installed and can't have dev paths scattered all over my system due to version conflicts. I generally keep everything in a separate path under C:\Dev\Python39-x86\;C:\Dev\Python39-x64\ for example.
>
I have cleared(Deleted) the created \Roaming\* Code and Python directories and tested several times with the same result.
Windows 10 Pro,19043.1151, x64

**Comment by stonebig** [comment]
Do you launch vscode via the winpython provided icon short cut?

**Comment by stonebig** [comment]
That said, If winpython doesn't seem to fit your environment , you may try with  another distribution.

**Comment by stonebig** [comment]
I read august v2021 version of python-vscode may have change things  for good or bad

Https://twitter.com/pythonvscode/status/1423710158891352066?s=19

**Comment by Axle-Ozz-i-sofT** [comment]
Sorry, Was Crashing my way through learning Python :)
I attempted both of these twice.
\WPy64-3940\VS Code.exe
\WPy64-3940\t\VSCode\Code.exe
with the same result.
C:\Users\name\AppData\Roaming\Python\Python39
>
But on checking I have just noticed that another 1 IDEs (Thonny) are doing the same thing to me and installing packages and dependencies to:
C:\Users\Axle\AppData\Roaming\Python\Python37
Even though I don't have version 3.7.x installed, and my V 3.9.x is set with the path under C:\Python39\
It too was installed as the "Portable" version :(
Thonny is built on V 3.7 and ships with it's own python version under the Thonny Application directory, so I'm expecting additional libraries and dependencies to install to the python directories and IDE current path.
Maybe it's just a python thing ¯\_(ツ)_/¯ I have attempted to get over a dozen apps to run from of PyPi via pip, and not one success yet. Many were missing dependencies if I went off the imports, but the apps(Under Scripts. site-packages) didn't list any dependencies that reconcile to from_Import + even minimal documentation standard on anything from PyPi appears to be Zero.
PyScripter is working fine off of the standard python V3.9.6 library ¯\_(ツ)_/¯
Don't stress too much, sooner I can get past this Python module and go back to C/C++ the better lol

**Comment by stonebig** [comment]
I agree things have gone south since the days of the [Quiche Eaters](https://www.ee.ryerson.ca/~elf/hack/realmen.html)

Will try later to understand your bug with multiple inheritance origin

**Comment by Axle-Ozz-i-sofT** [comment]
Lol, haven't hit that page in a hundred years. Thanks for the smile.
I'm doing this to help my lecturer create a beginners Python learner module. He does Java and is blind... because he can't C Boom Boom.
Seriously though, he was advised to go with Thonny, which has limitations that don't impress me, I have found PyScripter to be a sweet no bells IDE and it works, so I just have to see why pip and PyPi apps are not working. Some people seam to like the whole bells and whistles VSCode VS studio thing, but I prefer simple (text editor with some highlighting, syntax check and a couple of conveniences to point the workflow at the correct path and debugger )
I will come back to WinPython for another look, but for the moment I just need to learn python fundamentals and get this learner guide done :)
ty
